resources/schema.py

```diff
@@ -328,6 +348,62 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         return DeleteBerthTypeMutation()
 
 
+class CreateHarborMutation(graphene.ClientIDMutation):
+    class Input:
+        area = HarborAreaInput(required=True)
+        municipality = graphene.String()
+        image_file = graphene.String()
+        availability_level = graphene.ID()
+        number_of_places = graphene.Int()
+        maximum_width = graphene.Int()
+        maximum_length = graphene.Int()
+        maximum_depth = graphene.Int()
+        name = graphene.String()
+        street_address = graphene.String()
+
+    harbor = graphene.Field(HarborNode)
+
+    @classmethod
+    @login_required
+    @superuser_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO: Should check if the user has permissions to
+        # delete the specific resource
+        lang = info.context.META.get("HTTP_ACCEPT_LANG", settings.LANGUAGE_CODE).split(
```

Comment:
hm, trying to track down which of our package checks it where and how, maybe using just django's [`get_language()`](https://docs.djangoproject.com/en/3.0/_modules/django/utils/translation/#get_language) would work here? could use test it? I would like to somehow avoid `get`ting from `META` 😅

Reply:
By doing it from `META`, we're getting the language coming from the request. If I get it right, `get_language` would only tell you the default language of the running Django. But I'll take a look as well.

Reply:
yeah, I was thinking that since graphene can handle that request header, some package somewhere must be checking for it, and I was trying to figure out where it happens and how it does it. if they also do `META.get` then I guess there is no prettier way to do it, but maybe there is. 😅  I could not find where that check happens in graphene however, so I thought maybe it's just at the Django level and maybe `get_language` would work for this. maybe it doesn't, so this needs slightly more tinkering 🙂

Reply:
Django also has `get_language_from_request`

Reply:
😍 probably that is what some package uses somewhere, which I could not find 👍

Reply:
That said, I made a view:

```python
def language_check(request, *args, **kwargs):
    print(get_language())
    print(get_language_from_request(request))
    return HttpResponse(status=200)
```

And both `get_language` and `get_language_from_request` seemed to be following the language information my browser was sending i.e. `get_language` should probably work in this case. Of course you also need to have `django.middleware.locale.LocaleMiddleware` middleware configured, which you do.

Reply:
but probably [`get_language`](https://github.com/django/django/blob/stable/2.2.x/django/utils/translation/trans_real.py#L231) would not catch the `HTTP_ACCEPT_LANG` header, while `get_language_from_request` seems to [do that](https://github.com/django/django/blob/stable/2.2.x/django/utils/translation/trans_real.py#L474)

Reply:
That is true. Though looks like `get_language` will return the active translation, which is changed by [LocaleMiddleware](https://github.com/django/django/blob/stable/2.2.x/django/middleware/locale.py#L18-L26). `LocaleMiddleware` itself is using `get_language_from_request` for extracting the language from the request.

Reply:
ah, true! so, `get_language` should work for our purposes 👍